File: 1071.cpp
// Introductory Problems
// https://cses.fi/problemset/task/1071/

#include <iostream>
using namespace std;
typedef long long ll;

int main() {
  int t, r, c;
  cin >> t;
  while (t--) {
    cin >> r >> c;
    int a = max(r, c);
    int b = min(r, c);
    ll s = ll(a - 1) * (a - 1);
    if (a == b) s += a;
    else if (a % 2) {
      if (c == a) s+= a * 2 - r;
      else s += c;
    } else {
      if (r == a) s += a * 2 - c;
      else s += r;
    }
    cout << s << endl;
  }
}

File: 1071.py
# Introductory Problems : Number Spiral
#  https://cses.fi/problemset/task/1071/

# observe grid, generalize a formula.

for _ in range(int(input())):
    r, c = map(int, input().split())
    if c > r:
        if c%2==0:
            c-=1
            print((c**2+r))
        else:
            print(((c**2)-r+1))
    else:
        # now the formula gets reversed
        if r%2==0:
            print(((r**2)-c+1))
        else:
            r-=1
            print((r**2+c))